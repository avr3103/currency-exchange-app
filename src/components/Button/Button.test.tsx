import React from 'react';
import { render } from '@testing-library/react';
import { Button } from './index';

const handleClick = jest.fn();

it('renders Button', () => {
  const { asFragment } = render(<Button isActive={true} onClick={handleClick} />);
  expect(asFragment()).toMatchSnapshot();
});

it('Button must be disabled', () => {
  const { asFragment } = render(<Button isActive={false} onClick={handleClick} />);
  expect(asFragment().firstChild).toHaveClass('disabled');
});

it('Button must be enabled', () => {
  const { asFragment } = render(<Button isActive={true} onClick={handleClick} />);
  expect(asFragment().firstChild).not.toHaveClass('disabled');
});




