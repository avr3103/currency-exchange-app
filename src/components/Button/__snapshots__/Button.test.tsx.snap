// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button 1`] = `
<DocumentFragment>
  <button
    class="btn undefined "
  />
</DocumentFragment>
`;
